utils/material_sheet_keys.py

```diff
@@ -0,0 +1,249 @@
+
+    return all_sheets_dict
+
+
+def get_preferred_unit(row):
+    """
+    checks if there is a preferred unit
+    if no preferred unit then it uses the SI unit
+    if no preferred unit or SI unit, then it returns an empty string
+
+    :params row: pandas series
+    :returns: string unit or an empty string
+    """
+
+    # Guard clause: handling a case that I pass in a row from a sheet
+    # that doesn't have any units' column,in that case just tell me and do nothing
+    if "Preferred unit" not in row and "SI unit" not in row:
+        return
+
+    # making them into variables, so I don't have to keep repeating them
+    preferred_unit = row["Preferred unit"]
+    si_unit = row["SI unit"]
+
+    # has a preferred unit ie not "", "None", nor nan, then return preferred_unit
+    if (
+        preferred_unit != ""
+        and preferred_unit != "None"
+        and not pd.isna(preferred_unit)
+    ):
+        return preferred_unit
+
+    # does not have preferred unit, but has SI unit
+    elif (
+        preferred_unit == "" or preferred_unit == "None" or pd.isna(preferred_unit)
+    ) and (si_unit != "" and si_unit != "None" and not pd.isna(si_unit)):
+        return si_unit
+
+    # si unit and preferred unit are both empty or None
+    elif (
+        preferred_unit == "" or preferred_unit == "None" or pd.isna(preferred_unit)
+    ) and (si_unit == "" or si_unit == "None" or pd.isna(si_unit)):
+        return ""
+
+    # if somehow all the if cases are passed,
+    # then I need to know about it and create new conditions to capture them
+    else:
+        raise Exception(
+            f'hit else for {row["Name"]}; preferred_unit: {preferred_unit}; si_unit: {si_unit}'
+        )
+
+
+def get_instructions(row, sheet_name):
+    """
+    takes a row and gives back the instructions
+
+    :params row: padnas series
+    :returns instructions: a string that represents the instructions for that column
+    """
+
+    # gave the same instruction for all relation fields
+    # e.g.: "Pick value from *name column of data sheet"
+    if sheet_name == "relation":
+        instruction = f"Pick value from *name column of {row['Name']} sheet"
+        return instruction
+
+    if (sheet_name == "attribute") and (row["Name"] == "uncertainty_type"):
+        return "Pick uncertainty type from criptapp.org controlled vocabulary"
+
+    # guard clause that returns empty string, if there is no description column in that sheet
+    if "Description" not in row:
+        return ""
+
+    # if "Description" column exists, then instruction is description column
+    if "Description" in row:
+        return row["Description"]
+
+
+def get_new_df():
+    """
+    creates a new DF with columns needed for the Excel file options
+    returns: pandas dataframe object
+    """
+    category = "category"
+    column_name = "Name"
+    unit = "unit"
+    instructions = "instructions"
+
+    df = pd.DataFrame(columns=[category, column_name, unit, instructions])
```

Comment:
Creating variables for these names is a bit redundant

Reply:
I thought those would improve code readability and if we ever needed to change things it would be a bit easier, but I think you're right and it's not needed.